content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java

```diff
@@ -2289,80 +2304,39 @@ private String getTEM(String cid) {
-
-		Map togo = new HashMap();
-		if (!turnitinConn.isUseSourceParameter()) {
-			togo.put("uem", turnitinConn.getDefaultInstructorEmail());
-			togo.put("ufn", turnitinConn.getDefaultInstructorFName());
-			togo.put("uln", turnitinConn.getDefaultInstructorLName());
-			togo.put("uid", turnitinConn.getDefaultInstructorId());
-		} else {
-			String INST_ROLE = "section.role.instructor";
-			User inst = null;
-			try {
-				Site site = siteService.getSite(siteId);
-				User user = userDirectoryService.getCurrentUser();
+	public Map<String, String> getInstructorInfo(final String siteId, final boolean ignoreUseSource) {
+		log.debug("Getting instructor info for site {}", siteId);
 
-				log.debug("Current user: " + user.getId());
-
-				if (site.isAllowed(user.getId(), INST_ROLE)) {
-					inst = user;
-				} else {
-					Set<String> instIds = getActiveInstructorIds(INST_ROLE, site);
-					if (instIds.size() > 0) {
-						inst = userDirectoryService.getUser((String) instIds.toArray()[0]);
-					}
-				}
-			} catch (IdUnusedException e) {
-				log.error("Unable to fetch site in getAbsoluteInstructorInfo: " + siteId, e);
-			} catch (UserNotDefinedException e) {
-				log.error("Unable to fetch user in getAbsoluteInstructorInfo", e);
-			}
+		Map<String, String> togo = new HashMap<>();
 
-			if (inst == null) {
-				log.error("Instructor is null in getAbsoluteInstructorInfo");
-			} else {
-				togo.put("uem", getEmail(inst));
-				togo.put("ufn", inst.getFirstName());
-				togo.put("uln", inst.getLastName());
-				togo.put("uid", inst.getId());
-				togo.put("username", inst.getDisplayName());
-			}
+		// Check cache to avoid hitting database
+		if (instructorInfoCache.containsKey(siteId)) {
+			return instructorInfoCache.get(siteId);
 		}
 
-		return togo;
-	}
-
-	@SuppressWarnings("unchecked")
-	public Map getInstructorInfo(String siteId, boolean ignoreUseSource) {
-		Map togo = new HashMap();
-		if (!turnitinConn.isUseSourceParameter() && ignoreUseSource == false) {
+		if (!turnitinConn.isUseSourceParameter() && !ignoreUseSource) {
 			togo.put("uem", turnitinConn.getDefaultInstructorEmail());
 			togo.put("ufn", turnitinConn.getDefaultInstructorFName());
 			togo.put("uln", turnitinConn.getDefaultInstructorLName());
 			togo.put("uid", turnitinConn.getDefaultInstructorId());
 		} else {
-			String INST_ROLE = "section.role.instructor";
 			User inst = null;
 			try {
 				Site site = siteService.getSite(siteId);
 				User user = userDirectoryService.getCurrentUser();
+
+				log.debug("Current user: {}", user.getId());
+
 				if (site.isAllowed(user.getId(), INST_ROLE)) {
 					inst = user;
 				} else {
-					Set<String> instIds = getActiveInstructorIds(INST_ROLE, site);
-					if (instIds.size() > 0) {
-						inst = userDirectoryService.getUser((String) instIds.toArray()[0]);
+					Set<User> insts = getActiveInstructorIds(INST_ROLE, site);
+					if (!insts.isEmpty()) {
+						inst = insts.stream().findFirst().get();
```

Comment:
findFirst will return all Users then sort them and return the first, or are you looking just to find any?